lib/phoenix_component.ex

```diff
@@ -1149,7 +1149,11 @@ defmodule Phoenix.Component do
   end
 
   def __render_slot__(changed, entry, argument) when is_map(entry) do
-    entry.inner_block.(changed, argument)
+    if is_struct(entry.inner_block, Phoenix.LiveView.Rendered) do
+      entry.inner_block
+    else
+      entry.inner_block.(changed, argument)
+    end
```

Comment:
Minor nitpick but pattern matching should be more efficient (this is definitely a micro-optimization / bit shaving). But I also think it is clearer?

```suggestion
    case entry.inner_block do
      %Phoenix.LiveView.Rendered{} = rendered -> rendered
      fun -> fun.(changed, argument)
    end
```

Reply:
I agree that it is cleaner!